kernel/src/path.rs

```diff
@@ -1,34 +1,108 @@
 //! Utilities to make working with directory and file paths easier
 
-use lazy_static::lazy_static;
-use regex::Regex;
 use url::Url;
 
 use crate::{DeltaResult, Version};
 
-/// The delimiter to separate object namespaces, creating a directory structure.
+/// The delimiter to separate object namespaces, creating a directory structure. Note this is in url
+/// terms, so we use `/`
 const DELIMITER: &str = "/";
 
-lazy_static! {
-    static ref CHECKPOINT_FILE_PATTERN: Regex =
-        Regex::new(r"\d+\.checkpoint(\.\d+\.\d+)?\.parquet").unwrap();
-    static ref DELTA_FILE_PATTERN: Regex = Regex::new(r"\d+\.json").unwrap();
+#[derive(Debug)]
+pub(crate) struct LogPath<'a> {
+    url: &'a Url,
+    path: &'a str,
+    version: Option<Version>,
+    // if is compacted, this path spans version [`version`, `compacted_to_version`]
+    _compacted_to_version: Option<Version>,
```

Comment:
Why underscore?

Reply:
Not used at the moment, but seems like something we will likely want in the future, so I've left it in.